src/uct/tcp/tcp_ep.c

```diff
@@ -1066,7 +1351,8 @@ ucs_status_t uct_tcp_ep_flush(uct_ep_h tl_ep, unsigned flags,
 {
     uct_tcp_ep_t *ep = ucs_derived_of(tl_ep, uct_tcp_ep_t);
 
-    if (uct_tcp_ep_check_tx_res(ep) == UCS_ERR_NO_RESOURCE) {
+    if (ep->outstanding ||
```

Comment:
we need to return `UCS_IN_PROGRESS` in case of `ep->outstanding`

Reply:
done